server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/v2/DialectProcessingService.java

```diff
@@ -147,4 +150,29 @@ private List<Node> processDocument(
     }
     return nodes;
   }
+
+  private Collection<SyntaxError> createErrors(
+      ExtendedDocument document, String copybookId, DiagnosticPayload[] diagnostics) {
+    List<SyntaxError> errorList = new ArrayList<>();
+    for (DiagnosticPayload diagnostic : diagnostics) {
+      Location location = document.mapLocation(diagnostic.getRange());
+      errorList.add(
+          DialectErrorHelper.dialectError(
+              Locality.builder()
+                  .copybookId(copybookId)
+                  .uri(location.getUri())
+                  .range(location.getRange())
+                  .build(),
+              diagnostic.getMessage(),
+              getErrorCode(diagnostic.getErrorCode())));
```

Comment:
This does not make much sense to me. I can provide exactly one error code with arbitrary message text?